contracts/ronin/profile/ProfileStorage.sol

```diff
@@ -22,14 +24,69 @@ abstract contract ProfileStorage is IProfile, HasContracts {
     _profile.governor = newProfile.governor;
     _profile.pubkey = newProfile.pubkey;
 
+    _registry[uint256(uint160(newProfile.id))] = true;
+    _registry[uint256(uint160(newProfile.consensus))] = true;
+    _registry[uint256(uint160(newProfile.admin))] = true;
+    _registry[uint256(uint160(address(newProfile.treasury)))] = true;
+    _registry[uint256(uint160(newProfile.governor))] = true;
+    _registry[_hashPubkey(newProfile.pubkey)] = true;
+
     emit ProfileAdded(newProfile.id);
   }
 
   /**
-   * @dev Get an existed profile struct from id. Revert if the profile does not exists.
+   * @dev Get an existed profile struct from `id`. Revert if the profile does not exists.
    */
   function _getId2ProfileHelper(address id) internal view returns (CandidateProfile storage _profile) {
     _profile = _id2Profile[id];
     if (_profile.id == address(0)) revert ErrNonExistentProfile();
   }
+
+  /**
+   * @dev Checks each element in the new profile and reverts if there is duplication with any existing profile.
+   */
+  function _checkDuplicatedInRegistry(CandidateProfile memory profile) internal view {
+    _checkDuplicatedConsensus(profile.consensus);
+    _checkDuplicatedAdmin(profile.admin);
+    _checkDuplicatedTreasury(profile.treasury);
+    _checkDuplicatedGovernor(profile.governor);
+    _checkDuplicatedPubkey(profile.pubkey);
+  }
+
+  function _checkDuplicatedConsensus(address consensus) internal view {
+    if (_registry[uint256(uint160(consensus))]) {
+      revert ErrDuplicatedInfo("consensus", uint256(uint160(consensus)));
+    }
+  }
+
+  function _checkDuplicatedAdmin(address admin) internal view {
+    if (_registry[uint256(uint160(admin))]) {
+      revert ErrDuplicatedInfo("admin", uint256(uint160(admin)));
+    }
+  }
+
+  function _checkDuplicatedTreasury(address treasury) internal view {
+    if (_registry[uint256(uint160(treasury))]) {
+      revert ErrDuplicatedInfo("treasury", uint256(uint160(treasury)));
+    }
+  }
+
+  function _checkDuplicatedGovernor(address governor) internal view {
+    if (_registry[uint256(uint160(governor))]) {
+      revert ErrDuplicatedInfo("governor", uint256(uint160(governor)));
+    }
+  }
+
+  function _checkDuplicatedPubkey(bytes memory pubkey) internal view {
```

Comment:
Missing comments

Reply:
I consider the comments scary noise comment (See Page 64 of [this book](https://books.google.com.vn/books?redir_esc=y&id=_i6bDeoCQzsC&q=noise+comment#v=onepage&q&f=false)), so I intentionally left it blank.